### PR TITLE
Book Library Page BE

### PR DESCRIPTION
**Pull request by asmaa-ezz**
## Description

convert static view to dynamic for library books
add function in back-end helbers to status book in library

### Type of change

New features #35 
Description :
add function in back-end helbers to status book in library

New features #16
Description :
add queries in database getLibraryBooks

### CODE REVIEW

File: src/controllers/books.js [hunk 1]
```diff
@@ -1,13 +1,24 @@
...
+          layout: 'admin',
+          title: 'عرض المكتبة',
+          style: 'book',
+          js: 'book',
+          admin: 'admin',
+          resLibraryBooks,
+          status,
+        });
+    })
+    .catch(err => console.log(err));
```
Comment (ali-7) [inline src/controllers/books.js]: remove ```console.log(err)```  it should be ```next(err)```

**Review by ali-7 — CHANGES_REQUESTED** [review]

**Review by lubnaabd — APPROVED** [review]

File: src/controllers/books.js [hunk 2]
```diff
@@ -1,13 +1,24 @@
...
-      book: 'active',
-      layout: 'admin',
-      title: 'الكتب',
-      style: 'book',
-      js: 'book',
-    });
+const { getLibraryBooks } = require('../database/queries/book');
+const { status } = require('../views/helpers/index');
+
+
```
Comment (RamyAlshurafa) [inline src/controllers/books.js]: remove this line

File: src/database/queries/test/test.js [hunk 3]
```diff
@@ -0,0 +1,23 @@
+const test = require('tape');
+const runDbBuild = require('../../db_build');
+const { getLibraryBooks } = require('../book');
+
+
```
Comment (RamyAlshurafa) [inline src/database/queries/test/test.js:68]: remove this line

File: src/views/helpers/status.js [hunk 4]
```diff
@@ -0,0 +1,9 @@
+module.exports = (num) => {
+  let statusString = '';
```
Comment (RamyAlshurafa) [inline src/views/helpers/status.js]: what about
```js
module.exports = (num) =>  (num === 0) ? 'متاح' : 'مستعار'
```
you are free to use it or not

File: src/database/queries/test/test.js [hunk 5]
```diff
@@ -0,0 +1,23 @@
...
+test('Test getLibraryBooks', (t) => {
+  runDbBuild('db_bulid.sql', (err, res) => {
+    t.notOk(err);
+    return runDbBuild('fake_data.sql', () => {
+      getLibraryBooks()
+        .then((response) => {
+          t.equal(response.length > 0, true, 'successfully');
```
Comment (RamyAlshurafa) [inline src/database/queries/test/test.js]: use `response.length === 5` (`5` as example) insted of `response.length > 0`  
you know the books number in the `fake_data` so be specific

**Review by RamyAlshurafa — CHANGES_REQUESTED** [review]
keep your eye on indentation
**where is `controller/index`**

File: src/controllers/test/test.js [hunk 6]
```diff
@@ -76,6 +76,7 @@ test('test for website landing page route ', (t) => {
     });
 });
 
+
```
Comment (RamyAlshurafa) [inline src/controllers/test/test.js:140]: remove this line

**Review by RamyAlshurafa — CHANGES_REQUESTED** [review]

**Review by ali-7 — APPROVED** [review]

**Comment by RamyAlshurafa** [comment]
Don't request changes until you finished all requested changes
@Asmaa-ezz

**Review by RamyAlshurafa — DISMISSED** [review]

File: src/controllers/test/test.js [hunk 7]
```diff
@@ -245,4 +244,24 @@ test('test for add categery page route - with cookie and auth ', (t) => {
...
+        t.error(err);
+      }
+      t.equal(res.res.statusMessage, 'OK', 'statusMessage should return OK');
+      t.equal(typeof res.body, 'object', 'add categery returns data successfully ');
+      t.equal(res.text.includes('<title>عرض المكتبة</title>'), true, 'the page should have title \'الرئيسية\'');
+      t.end();
+    });
+});
+
+
```
Comment (RamyAlshurafa) [inline src/controllers/test/test.js]: remove this line

File: public/css/book.css [hunk 8]
```diff
@@ -120,28 +120,35 @@
...
+.link {
   color: white;
   border: 0px;
   padding-left: 0.5rem;
   padding-right: 0.5rem;
   font-family: cairo;
   border-radius: 5px;
   box-shadow: 1px 28px 35px -20px rgba(0, 0, 0, 0.75);
+  text-decoration: none;
+  display: inline-block;
```
Comment (RamyAlshurafa) [inline public/css/book.css]: add a line before each styling block

File: public/css/book.css [hunk 9]
```diff
@@ -120,28 +120,35 @@
...
 }
 .button--add {
```
Comment (RamyAlshurafa) [inline public/css/book.css:135]: add a line before each styling block

File: public/css/book.css [hunk 10]
```diff
@@ -120,28 +120,35 @@
...
   float: left;
 }
 
-.button--view {
+.link--view {
   background-color: #00c0ef;
 }
 
-.button--delete {
+
```
Comment (RamyAlshurafa) [inline public/css/book.css]: remove this line

File: src/controllers/test/test.js [hunk 11]
```diff
@@ -245,4 +244,24 @@ test('test for add categery page route - with cookie and auth ', (t) => {
     });
 });
 
+// library view page
+
```
Comment (RamyAlshurafa) [inline src/controllers/test/test.js]: remove this line

**Review by RamyAlshurafa — CHANGES_REQUESTED** [review]
## Indentation

**Review by RamyAlshurafa — APPROVED** [review]
Good Job :+1:

**Pull request merged**